samples/batch/transcription-enabled-storage/CreateByTimerFunction/CreateTranscriptionByTimer.cs

```diff
@@ -0,0 +1,348 @@
+            {
+                throw new ArgumentNullException(nameof(myTimer));
+            }
+
+            Logger = log;
+            StartDateTime = DateTime.UtcNow;
+            log.LogInformation($"C# Timer trigger function v3 executed at: {StartDateTime}. Next occurrence on {myTimer.Schedule.GetNextOccurrence(StartDateTime)}.");
+
+            if (!string.IsNullOrEmpty(CreateTranscriptionEnvironmentVariables.SecondaryLocale) && !CreateTranscriptionEnvironmentVariables.SecondaryLocale.Equals("None", StringComparison.OrdinalIgnoreCase))
+            {
+                log.LogWarning($"Language identification is not supported for timer-based transcription creation.");
+            }
+
+            var receiver = new MessageReceiver(CreateTranscriptionEnvironmentVariables.AzureServiceBus, "create_transcription_queue");
+
+            var totalServiceBusMessages = new List<Message>();
+
+            var serviceBusMessagesRemaining = true;
+            log.LogInformation("Pulling messages from queue...");
+            do
+            {
+                var messages = await receiver.ReceiveAsync(MaxMessagesPerRequest, TimeSpan.FromSeconds(20)).ConfigureAwait(false);
+
+                if (messages == null)
+                {
+                    log.LogInformation("No remaining messages found in queue.");
+                    break;
+                }
+
+                foreach (var message in messages)
+                {
+                    await receiver.CompleteAsync(message.SystemProperties.LockToken).ConfigureAwait(false);
+                    totalServiceBusMessages.Add(message);
+                }
+
+                if (messages.Count < MaxMessagesPerRequest)
+                {
+                    serviceBusMessagesRemaining = false;
+                }
+            }
+            while (serviceBusMessagesRemaining);
+
+            log.LogInformation($"Received {totalServiceBusMessages.Count} messages in total from queue.");
+
+            var validMessages = totalServiceBusMessages.Where(message => IsValidServiceBusMessage(message));
+
+            if (!validMessages.Any())
+            {
+                log.LogInformation("No valid messages were found in this function execution.");
+                return;
+            }
+
+            var validMessageCount = validMessages.Count();
+
+            log.LogInformation($"{validMessageCount} valid messages were found in this function execution.");
+            var chunkedMessages = new List<List<Message>>();
+
+            for (int i = 0; i < validMessageCount; i += MaxFilesPerTranscriptionJob)
+            {
+                var chunk = validMessages.Skip(i).Take(Math.Min(MaxFilesPerTranscriptionJob, validMessageCount - i)).ToList();
+                chunkedMessages.Add(chunk);
+            }
+
+            for (var i = 0; i < chunkedMessages.Count; i++)
+            {
+                await StartTranscriptionJob(chunkedMessages.ElementAt(i), log, i).ConfigureAwait(false);
```

Comment:
Why not call StartTranscriptionJob directly inside the loop L99-L102, so no need to create chnunkMessages

Reply:
because some messages inside of the service bus could be invalid. If in theory 90 % of all messages are invalid and our limit for files per job would be 100, we would only create transcription jobs with 10 files each. But if we create the jobs only on the valid messages, it'll always be 100 files per job